File: features/steps/web_steps.py
import subprocess
from pathlib import Path

from behave import given, when, then

WEB_DIR = Path(__file__).resolve().parents[2] / "web"
RUN_SCRIPT = "run_sequence.js"


def _run_sequence(sequence: str) -> str:
    result = subprocess.run(
        ["node", RUN_SCRIPT, sequence],
        cwd=WEB_DIR,
        capture_output=True,
        text=True,
        check=True,
    )
    return (result.stdout or "").strip() or "0"


@given("que a calculadora web foi carregada")
def step_load(context):
    context.sequence = ""
    context.display = "0"


@when('eu digito "{sequencia}"')
def step_type(context, sequencia: str):
    context.sequence = sequencia
    context.display = _run_sequence(sequencia)


@then('eu vejo "{esperado}" no visor')
def step_check(context, esperado: str):
    assert context.display == esperado, f"Esperado {esperado}, mas estava {context.display}"

File: features/steps/web.steps.js
import assert from "node:assert/strict";
import { Given, When, Then } from "@cucumber/cucumber";
import { createCalculator } from "../../src/calculator.js";

Given("que a calculadora web foi carregada", function () {
  this.calculator = createCalculator();
  this.display = this.calculator.getDisplay();
});

When("eu digito {string}", function (sequencia) {
  for (const symbol of sequencia) {
    if (symbol.trim() === "") {
      continue;
    }
    this.display = this.calculator.press(symbol);
  }
});

Then("eu vejo {string} no visor", function (esperado) {
  assert.equal(this.display, esperado);
});
